src/units.cpp

```diff
@@ -126,43 +140,118 @@ std::vector<Unit>::iterator Units::UnitsImpl::findUnit(const std::string &refere
                         [=](const Unit &u) -> bool { return u.mReference == reference; });
 }
 
-Units::Units()
-    : mPimpl(new UnitsImpl())
+/*
+
+//TDOD: This code is progress towards checking that units are compatible.
+using UnitMultiplierMap = std::map<std::string, double>;
+
+    UnitMultiplierMap unitMap = {};
+    for (const auto &baseUnits : baseUnitsList) {
+        unitMap[baseUnits] = 0.0;
+    }
+bool updateUnitMultipliers(UnitMultiplierMap &unitMultiplierMap,
+                           double &multiplier,
+                           const UnitsPtr &units,
+                           double uExp, double logMult,
+                           int direction)
 {
+    bool updated = false;
+    auto unitsName = units->name();
+    if (!units->isBaseUnit()) {
+        std::string ref;
+        std::string pre;
+        std::string id;
+        double exp;
+        double mult;
+        double expMult;
+        for (size_t i = 0; i < units->unitCount(); ++i) {
+            units->unitAttributes(i, ref, pre, exp, expMult, id);
+            mult = std::log10(expMult);
+            if (!isStandardUnitName(ref)) {
+                auto model = owningModel(units);
+                auto refUnits = model->units(ref);
+                updated = updateUnitMultipliers(unitMultiplierMap, multiplier, refUnits, exp * uExp, logMult + mult * uExp + standardPrefixList.at(pre) * uExp, direction);
```

Comment:
I haven't checked the code in details, but couldn't there be a case where `updated` was previously `true` and then `updateUnitMultipliers()` returns `false`, making `updated` `false`? Same question for the other `updateUnitMultipliers()` method.